java-checks/src/main/java/org/sonar/java/checks/helpers/ConstantUtils.java

```diff
@@ -84,7 +84,7 @@ public static Object resolveAsConstant(ExpressionTree tree) {
     if (expression.is(Tree.Kind.STRING_LITERAL)) {
       return LiteralUtils.trimQuotes(((LiteralTree) expression).value());
     }
-    if (tree.is(Tree.Kind.INT_LITERAL)) {
+    if (tree.is(Tree.Kind.INT_LITERAL, Tree.Kind.UNARY_MINUS, Tree.Kind.UNARY_PLUS)) {
```

Comment:
This will only work for Integer literals and not for Long literals. You can either make it work for Long literals, or you can add negative test, to document that it doesn't work for Long literals